lib/ansible/modules/cloud/amazon/cloudfront_distribution.py

```diff
@@ -0,0 +1,1993 @@
+            cache_behavior['forwarded_values'] = forwarded_values
+            return cache_behavior
+        except Exception as e:
+            self.module.fail_json_aws(e, msg="Error validating forwarded values")
+
+    def validate_lambda_function_associations(self, config, lambda_function_associations, cache_behavior):
+        try:
+            if lambda_function_associations is not None:
+                self.validate_is_list(lambda_function_associations, 'lambda_function_associations')
+                for association in lambda_function_associations:
+                    association = change_dict_key_name(association, 'lambda_function_arn', 'lambda_function_a_r_n')
+                    self.validate_attribute_with_allowed_values(association.get('event_type'), 'cache_behaviors[].lambda_function_associations.event_type',
+                                                                self.__valid_lambda_function_association_event_types)
+                cache_behavior['lambda_function_associations'] = ansible_list_to_cloudfront_list(lambda_function_associations)
+            else:
+                if 'lambda_function_associations' in config:
+                    cache_behavior['lambda_function_associations'] = config.get('lambda_function_associations')
+                else:
+                    cache_behavior['lambda_function_associations'] = ansible_list_to_cloudfront_list([])
+            return cache_behavior
+        except Exception as e:
+            self.module.fail_json_aws(e, msg="Error validating lambda function associations")
+
+    def validate_allowed_methods(self, config, allowed_methods, cache_behavior):
+        try:
+            if allowed_methods is not None:
+                self.validate_required_key('items', 'cache_behavior.allowed_methods.items[]', allowed_methods)
+                temp_allowed_items = allowed_methods.get('items')
+                self.validate_is_list(temp_allowed_items, 'cache_behavior.allowed_methods.items')
+                self.validate_attribute_list_with_allowed_list(temp_allowed_items, 'cache_behavior.allowed_methods.items[]',
+                                                               self.__valid_methods_allowed_methods)
+                cached_items = allowed_methods.get('cached_methods')
+                if 'cached_methods' in allowed_methods:
+                    self.validate_is_list(cached_items, 'cache_behavior.allowed_methods.cached_methods')
+                    self.validate_attribute_list_with_allowed_list(cached_items, 'cache_behavior.allowed_items.cached_methods[]',
+                                                                   self.__valid_methods_cached_methods)
+                # we don't care if the order of how cloudfront stores the methods differs - preserving existing
+                # order reduces likelihood of making unnecessary changes
+                if 'allowed_methods' in config and set(config['allowed_methods']['items']) == set(temp_allow
```

Comment:
This needs to have a default (`present` would be most consistent with other modules). If I have a play like:

```
- cloudfront_distribution:
    comment: This will not work
    alias: mysite.com
    default_origin_domain_name: foo.com
```

The task succeeds, but no distribution is created or updated, and there's no error message indicating that nothing happened.